### PR TITLE
Using Typesafe Config

### PR DESCRIPTION
**Pull request by akauppi**
Typesafe Config is best used via a class that gathers all access to such config. This makes access type safe, and makes sure the configuration is consistent (all keys exist in the `.conf` files).

I took the obvious cases into the code. Let's see the ones marked "config?" - please mark the ones you wish to be configurable and I shall edit the PR.

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-
-  actor {
-    debug {
-      # enable DEBUG logging of actor lifecycle changes
-      lifecycle = on
-      unhandled = on
-      autoreceive = on
-    }
-  }
-}
```
Comment (akauppi) [inline src/main/resources/application.conf]: This file is not needed. It's used in applications, but nakadi-klients is a library, thus providing a reference.conf instead.

File: src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala [hunk 2]
```diff
@@ -50,7 +50,7 @@ class KlientSupervisor private (val endpoint: URI, val port: Int, val securedCon
 
   var listenerMap: Map[String, ActorRef] = Map()
 
-  override val supervisorStrategy = AllForOneStrategy(maxNrOfRetries = 100, withinTimeRange = 5 minutes) {
+  override val supervisorStrategy = AllForOneStrategy(maxNrOfRetries = 100, withinTimeRange = 5 minutes) {    // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala]: I would say yes

File: src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala [hunk 3]
```diff
@@ -132,6 +132,6 @@ class KlientSupervisor private (val endpoint: URI, val port: Int, val securedCon
 
   def resolveActor(actorSelectionPath: String): Future[ActorRef] = {
     val receiverSelection = context.actorSelection(actorSelectionPath)
-    receiverSelection.resolveOne()(Timeout(1L, TimeUnit.SECONDS))
+    receiverSelection.resolveOne()(Timeout(1L, TimeUnit.SECONDS))     // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala]: I would say yes

File: src/main/scala/org/zalando/nakadi/client/actor/PartitionReceiver.scala [hunk 4]
```diff
@@ -71,7 +71,7 @@ class PartitionReceiver private (val endpoint: URI,
   var lastCursor: Option[Cursor] = None
   implicit val materializer = ActorMaterializer()
 
-  val RECEIVE_BUFFER_SIZE: Int = 1024
+  val RECEIVE_BUFFER_SIZE: Int = 1024   // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/PartitionReceiver.scala]: yes, please :-)

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala [hunk 5]
```diff
@@ -10,11 +10,12 @@ import de.zalando.scoop.{ScoopListener, ScoopClient}
 import org.slf4j.LoggerFactory
 import org.zalando.nakadi.client._
 import scala.concurrent.ExecutionContext.Implicits.global
+import org.zalando.nakadi.client.Conf
 
 object ScoopListenerActor{
-  val UNREACHABLE_MEMBER_EVENT_TYPE: String = "/scoop-system/unreachable-member"
-  val UNREACHABLE_MEMBER_EVENT_BODY_KEY: String = "unreachable_member"
-  val SCOOP_LISTENER_ID_PREFIX: String = "scoop-listener"
+  val UNREACHABLE_MEMBER_EVENT_TYPE: String = "/scoop-system/unreachable-member"    // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: no

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala [hunk 6]
```diff
@@ -10,11 +10,12 @@ import de.zalando.scoop.{ScoopListener, ScoopClient}
...
+  val UNREACHABLE_MEMBER_EVENT_BODY_KEY: String = "unreachable_member"              // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: no

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala [hunk 7]
```diff
@@ -10,11 +10,12 @@ import de.zalando.scoop.{ScoopListener, ScoopClient}
...
+  val SCOOP_LISTENER_ID_PREFIX: String = "scoop-listener"                           // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: yes

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala [hunk 8]
```diff
@@ -87,11 +88,11 @@ class ScoopListenerActor(listener: Listener,
   def handleMemberUnreachable(member: Member): Unit = {
     currentCluster match {
       case Some(actualCluster) =>
-        if(actualCluster.selfRoles.contains("leader")) { // TODO right role name?
+        if(actualCluster.selfRoles.contains("leader")) { // TODO right role name?   // config?
           logger.info("I AM the LEADER and I am notifying other Scoop aware clients about [unreachableMember={}]", member)
           val event = Event(UNREACHABLE_MEMBER_EVENT_TYPE,
             "scoop-system",
-            Map("id" -> UUID.randomUUID().toString),
+            Map("id" -> UUID.randomUUID().toString),      // @Benjamin: should that be the "id" coming from configuration? AKa110216
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: :+1:

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala [hunk 9]
```diff
@@ -87,11 +88,11 @@ class ScoopListenerActor(listener: Listener,
   def handleMemberUnreachable(member: Member): Unit = {
     currentCluster match {
       case Some(actualCluster) =>
-        if(actualCluster.selfRoles.contains("leader")) { // TODO right role name?
+        if(actualCluster.selfRoles.contains("leader")) { // TODO right role name?   // config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: hmm..not sure..I think it is unlikely the Akka cluster role name changes frequently

File: src/main/scala/org/zalando/nakadi/client/package.scala [hunk 10]
```diff
@@ -3,6 +3,9 @@ package org.zalando.nakadi
 
 package object client {
 
+  // @Benjamin: Do you want these to be coming from Config?
```
Comment (codechimp) [inline src/main/scala/org/zalando/nakadi/client/package.scala]: :-1:

File: src/test/java/org/zalando/nakadi/client/utils/Main.java [hunk 11]
```diff
@@ -25,7 +25,7 @@ public static void main(final String[] args) throws Exception {
 
         final Client client = new KlientBuilder()
                         .withEndpoint(new URI("localhost"))
-                        .withPort(8080)
+                        .withPort(8080)                         // test config?
```
Comment (codechimp) [inline src/test/java/org/zalando/nakadi/client/utils/Main.java]: :+1:

File: src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala (hunk 2 above)
Comment (akauppi) [inline src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala]: Akka is an interesting thing. I would kind of assume these to be configurable from within it, but last time I checked, they weren't. I envision there is some reason behind this, but am not sure. Will add to Config, tomorrow.

File: src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala (hunk 3 above)
Comment (akauppi) [inline src/main/scala/org/zalando/nakadi/client/actor/KlientSupervisor.scala]: ok

File: src/main/scala/org/zalando/nakadi/client/actor/PartitionReceiver.scala (hunk 4 above)
Comment (akauppi) [inline src/main/scala/org/zalando/nakadi/client/actor/PartitionReceiver.scala]: ok

File: src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala (hunk 7 above)
Comment (akauppi) [inline src/main/scala/org/zalando/nakadi/client/actor/ScoopListenerActor.scala]: What line is this refering to? The "scoop-listener" prefix?

**Pull request merged**